source/common/http/http1/codec_impl.h

```diff
@@ -399,7 +412,14 @@ class ServerConnectionImpl : public ServerConnection, public ConnectionImpl {
                        uint32_t max_request_headers_kb, const uint32_t max_request_headers_count,
                        envoy::config::core::v3::HttpProtocolOptions::HeadersWithUnderscoresAction
                            headers_with_underscores_action);
-
+  ~ServerConnectionImpl() {
+    // This ideally would not be necessary, however in the integration test framework the server
```

Comment:
Integration tests or unit tests?  I would think that integration tests would mimic envoy e2e behavior; triggering in e2e tests implies that this would also happen in production.

Reply:
integration tests.
in production, when the connection is closed, the L4 and L7 filter chains are all deleted.  In the integration tests we keep the fake server connection around so we can query it, and the only indication that the connection has been invalidated under the hood is SharedConnectionWrapper connected() is false.